coderd/files/cache.go

```diff
@@ -140,20 +141,34 @@ type cacheEntry struct {
 
 type fetcher func(context.Context, uuid.UUID) (CacheEntryValue, error)
 
+var _ fs.FS = (*CloseFS)(nil)
+var _ io.Closer = (*CloseFS)(nil)
+
+// CloseFS is a wrapper around fs.FS that implements io.Closer. The Close()
+// method tells the cache to release the fileID. Once all open references are
+// closed, the file is removed from the cache.
+type CloseFS struct {
+	fs.FS
+
+	close func() error
+}
+
+func (f *CloseFS) Close() error { return f.close() }
```

Comment:
I named this `Close` to match the pattern of opening a file and deferring a `Close`